src/server/routes/consents.ts

```diff
@@ -261,6 +269,226 @@ router.get(Routes.CONSENTS, loginMiddleware, (_: Request, res: Response) => {
   res.redirect(303, url);
 });
 
+//  ABTEST: followupConsent : Start
+// The following routes and functions are only used as part of the followupConsents AB test
+
+function getErrorResponse(
+  e: Error | IDAPIError,
+  state: RequestState,
+): [number, RequestState] {
+  let status;
+  let message;
+  if ('status' in e && 'error' in e) {
+    status = e.status;
+    message = e.error;
+  } else {
+    status = 500;
+    message = 'An error has occured, please try again.';
+  }
+  return [
+    status,
+    {
+      ...state,
+      globalMessage: {
+        ...state.globalMessage,
+        error: message,
+      },
+    },
+  ];
+}
+
+enum IDAPIEntity {
+  NEWSLETTERS = 'newsletters',
+  CONSENTS = 'consents',
+}
+type IDAPIEntityRecord = [IDAPIEntity, NewsLetter[] | Consent[]];
+type IDAPIEntityGetter = (
+  ip: string,
+  sc_gu_u: string,
+  geocode?: GeoLocation,
+) => Promise<IDAPIEntityRecord>;
+type IDAPIEntitySetter = (
+  ip: string,
+  sc_gu_u: string,
+  body: { [key: string]: string },
+  geocode?: GeoLocation,
+) => Promise<void>;
+
+function isSubscribed(entities: NewsLetter[] | Consent[]): boolean {
+  if (entities.length < 1) {
+    return false;
+  }
+  const entity = entities[0];
+  if ('subscribed' in entity) {
+    return entity?.subscribed ?? false;
+  }
+  if ('consented' in entity) {
+    return entity?.consented ?? false;
+  }
+  return false;
+}
+
+function getRedirectUrl(config: Configuration, state: RequestState): string {
+  return state?.queryParams?.returnUrl ?? config.defaultReturnUri;
+}
+
+async function getNewsletterEntity(
+  ip: string,
+  sc_gu_u: string,
+  geocode?: GeoLocation,
+): Promise<[IDAPIEntity, NewsLetter[]]> {
+  const newsletters = (
+    await getUserNewsletterSubscriptions(
+      NewsletterMap.get(geocode) as string[],
+      ip,
+      sc_gu_u,
+    )
+  ).slice(0, 1); // Assume 'Today' newsletter is the leading newsletter in NewsletterMap;
+  return [IDAPIEntity.NEWSLETTERS, newsletters];
+}
+
+async function getConsentEntity(
+  ip: string,
+  sc_gu_u: string,
+): Promise<[IDAPIEntity, Consent[]]> {
+  const consents = await getUserConsentsForPage(
+    [Consents.SUPPORTER],
+    ip,
+    sc_gu_u,
+  );
+  return [IDAPIEntity.CONSENTS, consents];
+}
+
+async function setNewsletterEntity(
+  ip: string,
+  sc_gu_u: string,
+  body: { [key: string]: string },
+  geocode?: GeoLocation,
+) {
+  const { update } = consentPages[1]; // Can reuse newsletter updater function.
```

Comment:
The access with a hardcoded index makes me feel uncomfortable. 

It might be worth pulling the definition of each page out of the array, then compose an array with these definitions. That way you can directly access the definition you fancy without having to care about the order of each page in the array.

Reply:
As discussed, I agree, but don't have the capacity to handle this refactor. It would be great to see this done in future though.